src/main/java/lai/util/TaskList.java

```diff
@@ -0,0 +1,101 @@
+package lai.util;
+
+import lai.task.Task;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Represents a list of tasks in the Lai application.
+ * This class provides functionality to add, remove, and access tasks,
+ * as well as to iterate over all tasks.
+ */
+public class TaskList implements Iterable<Task> {
+    protected List<Task> tasks = new ArrayList<>();
```

Comment:
It's not clear why the default value is needed since it's overwritten in every constructor.